TP3/Correction/vectors.py

```diff
@@ -0,0 +1,101 @@
+"""Manipulation of 2D-vectors.
+
+Note :
+``from __future__ import annotations`` est nécessaire si l'on veut pouvoir utiliser une classe
+en tant qu'annotation de type dans cette même classe ! Comme dans l'exemple suivant :
+class Point:
+    @classmethod
+    def origin(cls) -> Point:  # "Error: Point not defined" serait levé sans l'utilisation de ``from __future__ import annotations``
+
+"""
+
+from __future__ import annotations
+
+import math
+
+
+class Point:
+    def __init__(self, x: float, y: float) -> None:
+        self.x = x
+        self.y = y
+
+    @classmethod
+    def origin(cls) -> Point:
+        return Point(0, 0)
+
+    def __str__(self) -> str:
+        return f"Point({self.x}, {self.y})"
+
+
+class Vector:
+    def __init__(self, x: float, y: float) -> None:
+        self.x = x
+        self.y = y
```

Comment:
Un dernier point pour améliorer la compréhension, on pourrait utiliser des noms de parametres différents des noms d'attributs juste pour montrer qu'il n'y a aucune contrainte là-dessus:
 ```suggestion
    def __init__(self, x_coordinate: float, y_coordinate: float) -> None:
        self.x = x_coordinate
        self.y = y_coordinate
```